src/main/kotlin/net/leanix/vsm/sbomBooster/service/ProcessService.kt

```diff
@@ -16,34 +20,67 @@ class ProcessService(
 ) {
     companion object {
         private val logger: Logger = LoggerFactory.getLogger(ProcessService::class.java)
+        private var initStatus: Boolean = false
+    }
+
+    fun initOrt() {
+        runBlocking {
```

Comment:
I am new to Kotlin but have used parallel programming techniques in the past. The goal was to make sure that a second process didn't kick off the pull command while the first one was still pulling and hadn't updated the boolean flag yet. This `init` function is called from within the `Async` function `processRepository`, implying that the init function could be called more than once at a time. I'll remove the blocking scope just in case this isn't the correct way to handle the blocking, plus more than one pull isn't going to break anything, just a performance hit.

Reply:
I think that it will not work this way because there is no coroutine launched, but you have a very valid point on this. Maybe a simpler way is to fetch the latest image before starting the processing, just after getting the repos, ensuring that we use the same ORT version for processing all the repositories. 

Since multiple threads (based on the CONCURRENCY_FACTOR env variable) will have access to the `initStatus` variable via the `@Async` call of `processRepository`, if we need to use this variable, we need to "protect" it from concurrent access by multiple threads as well. 

In any case, calling docker command with `--pull=always` it's just a request to the registry to ensure that no new image is present. 

I am also new to Kotlin, since I come a long way from the Java world and coroutines is one of the areas that I haven't really explored yet.

Reply:
I removed the blocking scope so if it is all the same to you, I'd rather merge it this way and leave the changes you are suggesting as additional enhancements.